Add support for updating chart in response to period dimension selection

stevebaranski opened the pull request:
Fixes #11064.

To test:

- Checkout the branch, confirm that it builds & tests pass.
- For a site, navigate to Stats, and select a given period. Then select a period dimension (Views, Visitors, Likes, Comments.
- Please note that at the moment, the chart is the same regardless of period (i.e., Days, Weeks, Months, Years).
- Please also note that the fabricated numbers do not match the magnitude of the values in the stubbed JSON data.

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

![11064a_views_portrait](https://user-images.githubusercontent.com/221062/55040944-f6502d00-4fe7-11e9-8f18-b39a3b8f9fea.png)

![11064b_visitors_landscape](https://user-images.githubusercontent.com/221062/55040945-f6502d00-4fe7-11e9-926c-3ff0460f7166.png)

![11064c_likes_landscape](https://user-images.githubusercontent.com/221062/55040946-f6502d00-4fe7-11e9-8344-389946cbc3e7.png)

![11064d_comments_portrait](https://user-images.githubusercontent.com/221062/55040947-f6502d00-4fe7-11e9-874a-190776911e8e.png)

ScoutHarris commented on WordPress/Classes/ViewRelated/Stats/Charts/Charts+AxisFormatters.swift:
```diff
@@ -65,13 +65,44 @@ class VerticalAxisFormatter: IAxisValueFormatter {
+//            formattedValue = formatter.string(from: value) ?? "\(value)"
+//        }
+//
+//        return formattedValue
```
I'm assuming this block of commented out code can be removed?

stevebaranski replied:
Absolutely - thanks for catching that!

ScoutHarris commented:
Hey @stevebaranski . I noticed on the Visitors tab, the top of the y-axis is cut off just a smidge:

![y-axis](https://user-images.githubusercontent.com/1816888/55103016-48975980-508d-11e9-846b-04885717aab2.png)

stevebaranski commented:
Hey @ScoutHarris - thanks for catching my oversight(s). I have pushed two commits as requested:

1. Removed the commented-out code.
1. Adjusted the top offset slightly to prevent axis label clipping.

Here's a screenshot of the latter in action:
![11341pr_feedback](https://user-images.githubusercontent.com/221062/55108792-8d24f400-5091-11e9-839c-77bcf7e23687.png)

Do you mind taking another look?
